### PR TITLE
FIX: Remove lena

### PR DESCRIPTION
**Pull request by larsoner**
The `lena` image has an incompatible license. As discussed on the mailing list, this removes it with a hopefully informative error.

Examples and docs updated. Let me know if there's anything else.

### CODE REVIEW

**Comment by rgommers** [comment]
Link to the mailing list discussion: http://article.gmane.org/gmane.comp.python.scientific.devel/19898

File: scipy/misc/common.py [hunk 1]
```diff
@@ -356,41 +357,24 @@ def lena():
...
     Returns
     -------
-    lena : ndarray
-        Lena image
+    None
+
+    Raises
+    ------
+    RuntimeError
+        This functionality has been removed.
```
Comment (rgommers) [inline scipy/misc/common.py]: Maybe add here "due to licensing reasons".

**Comment by rgommers** [comment]
One more thing to be done: add this to the backwards-incompatible changes section of the 0.17.0 release notes. Otherwise looks good to me.

**Comment by Eric89GXL** [comment]
@rgommers done

**Comment by rgommers** [comment]
Merged. Thanks Eric.

**Pull request merged**